lib/util/sdp/index.js

```diff
@@ -251,6 +289,84 @@ function setSimulcast(sdp, sdpFormat, trackIdsToAttributes) {
   })).concat('').join('\r\n');
 }
 
+/**
+ * Filter local codecs based on the remote unified plan SDP.
+ * @param {string} localSdp
+ * @param {string} remoteSdp
+ * @returns {string} - Updated local SDP
+ */
+function unifiedPlanFilterLocalCodecs(localSdp, remoteSdp) {
+  const localMediaSections = getMediaSections(localSdp);
+  const localSession = localSdp.split('\r\nm=')[0];
+  const remoteMidsToMediaSections = createMidToMediaSectionMap(remoteSdp);
+  return [localSession].concat(localMediaSections.map(localSection => {
```

Comment:
@manjeshbhargav, this function is getting pretty big - please consider breaking it down further. one idea is to move  the mapping of localMediaSection -> targetSecion into another function 
```
return [localSession].concat(localMediaSections.map(localSection => convertLocalSection(localSection, remoteMidsToMediaSections)));
```

Reply:
Thanks for breaking down, It would be nice to get more unittest for the individual functions.